### PR TITLE
Fix parsing of line and column numbers in Polyspace parser

### PR DESCRIPTION
**Pull request by evahabeeballah**
Updated the parser file with Integer.valueOf() function for column and line set, since line number was showing as 0 for all issues in the dashboard. 
Also updated the logic of checking whether it is a BugFinder or a CodeProver file to be based on the header. (Test file edited accordingly) 

Relevant pull request : https://github.com/jenkinsci/warnings-ng-plugin/pull/1412

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/edu/hm/hafner/analysis/parser/PolyspaceParser.java [hunk 1]
```diff
@@ -44,18 +44,23 @@ private Report parse(final Stream<String> lines) {
             Report report = new Report();
             Iterator<String> lineIterator = lines.iterator();
 
+            String header = "";
+            if (lineIterator.hasNext()) {
+                header = lineIterator.next();
+            }
+
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/PolyspaceParser.java:51]: move into a method `readHeader`

File: src/main/java/edu/hm/hafner/analysis/parser/PolyspaceParser.java [hunk 2]
```diff
@@ -68,8 +73,8 @@ private Report parse(final Stream<String> lines) {
                     builder.setDescription(attributes[1]);
                     builder.setMessage("Check: " + attributes[5] + " " + attributes[6]);
                     builder.setModuleName(attributes[7]);
-                    builder.setColumnStart(attributes[colNumber]);
-                    builder.setLineStart(attributes[lineNumber]);
+                    builder.setColumnStart(Integer.parseInt(attributes[colNumber]));
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/PolyspaceParser.java:76]:
```suggestion
                    builder.setColumnStart(IntegerParser.parseInt(attributes[colNumber]));
```

**Review by uhafner — CHANGES_REQUESTED** [review]
Can you expose that bug in a test? Or is this already done by changing the log file?

File: src/main/java/edu/hm/hafner/analysis/parser/PolyspaceParser.java (hunk 1 above)
Comment (evahabeeballah) [reply src/main/java/edu/hm/hafner/analysis/parser/PolyspaceParser.java]: Done.

**Comment by evahabeeballah** [comment]
I couldn't expose this bug using the tests no, the assertions for line number (hasLineStart) in the test file were successful when the bug was there. I had to notice it in the dashboard only.

File: src/main/java/edu/hm/hafner/analysis/parser/PolyspaceParser.java (hunk 1 above)
Comment (uhafner) [reply src/main/java/edu/hm/hafner/analysis/parser/PolyspaceParser.java]: In Java you normally do not use variables in such a case, see next method.

**Comment by uhafner** [comment]
> I couldn't expose this bug using the tests no, the assertions for line number (hasLineStart) in the test file were successful when the bug was there. I had to notice it in the dashboard only.

Can you then add another snippet report that actually shows that this bug was present? It seems that only a single file is currently parsed, while the parser supports two different columns.

**Comment by evahabeeballah** [comment]
> > I couldn't expose this bug using the tests no, the assertions for line number (hasLineStart) in the test file were successful when the bug was there. I had to notice it in the dashboard only.
> 
> Can you then add another snippet report that actually shows that this bug was present? It seems that only a single file is currently parsed, while the parser supports two different columns.

Sorry I don't get what you mean about adding another snippet of the report to show that the bug was present. What i meant to say is that in the previous report snippet there was no example of an issue that will cause the bug to appear, so it was only discovered in the dashboard view.

**Comment by uhafner** [comment]
> > Can you then add another snippet report that actually shows that this bug was present? It seems that only a single file is currently parsed, while the parser supports two different columns.
> 
> Sorry I don't get what you mean about adding another snippet of the report to show that the bug was present. What i meant to say is that in the previous report snippet there was no example of an issue that will cause the bug to appear, so it was only discovered in the dashboard view.

This is exactly my point: you should only fix things that are visible via a test case. And your example in the existing tests contains only the header

```
ID	Family	Group	Color	New	Check	Information	Function	File	Status	Severity	Comment	CWE ID	Key	Line	Col
``` 

which is the first file format (`if`), but not the second file format (`else`). And the bug was located in the second format. So normally, one would expose that bug by adding another "polyspace.log" file that contains the second file format. Then you would have seen from the test that the column is not parsed correctly, *before* you are actually fixing it.

**Comment by evahabeeballah** [comment]
> > > Can you then add another snippet report that actually shows that this bug was present? It seems that only a single file is currently parsed, while the parser supports two different columns.
> > 
> > 
> > Sorry I don't get what you mean about adding another snippet of the report to show that the bug was present. What i meant to say is that in the previous report snippet there was no example of an issue that will cause the bug to appear, so it was only discovered in the dashboard view.
> 
> This is exactly my point: you should only fix things that are visible via a test case. And your example in the existing tests contains only the header
> 
> ```
> ID	Family	Group	Color	New	Check	Information	Function	File	Status	Severity	Comment	CWE ID	Key	Line	Col
> ```
> 
> which is the first file format (`if`), but not the second file format (`else`). And the bug was located in the second format. So normally, one would expose that bug by adding another "polyspace.log" file that contains the second file format. Then you would have seen from the test that the column is not parsed correctly, _before_ you are actually fixing it.

Thanks for the clarification, I updated the test with another file for polyspace for the "else" condition. Hope it is okay now.

**Review by uhafner — APPROVED** [review]

**Pull request merged**